src/ResourceManager/ApplicationInsights/Commands.ApplicationInsights/ApplicationInsights/NewApplicationInsightsContinuousExport.cs

```diff
@@ -0,0 +1,142 @@
+    public class NewAzureApplicationInsightsContinuousExportCommand : ApplicationInsightsBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Application Insights Component Object.")]
+        [ValidateNotNull]
+        public PSApplicationInsightsComponent ApplicationInsightsComponent { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Resource Id.")]
+        [ValidateNotNull]
+        public ResourceIdentifier ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Component Name.")]
+        [Alias(ApplicationInsightsComponentNameAlias, ComponentNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Document types that need exported.")]        
+        [ValidateSet(ApplicationInsightsBaseCmdlet.DocumentTypes.Requests,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.Exceptions,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.Event,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.Messages,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.Metrics,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.PageViewPerformance,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.PageViews,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.RemoteDependency,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.Availability,
+            ApplicationInsightsBaseCmdlet.DocumentTypes.PerformanceCounters,
+            IgnoreCase = true)]
+        public string[] DocumentType { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Destination Storage Account Id.")]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountId { get; set; }
+
+        [Parameter(
+            Position = 4,
```

Comment:
@frankguodongchen we try to cap the number of positional parameters a parameter set has at four - I would recommend making the core parameters (`ResourceGroupName`, `Name`, `ResourceId`, `InputObject`) the positional parameters and everything else isn't positional

Reply:
OK, updated.